internal/brokers/examplebroker.go

```diff
@@ -296,12 +297,21 @@ func (b *exampleBroker) IsAuthorized(ctx context.Context, sessionID, authenticat
 		}
 	}
 
-	// Handle authentication in a separate routine to allow for cancellation
-	cleanup, err := b.newIsAuthorizedCall(ctx, sessionID)
-	if err != nil {
-		return "", "", err
+	// Handles the context that will be assigned for the IsAuthorized handler
+	if _, exists := b.isAuthorizedCalls[sessionID]; exists {
+		return "", "", fmt.Errorf("IsAuthorized already running for session %q", sessionID)
 	}
-	defer cleanup()
+	cc, cancel := context.WithCancel(ctx)
```

Comment:
We are not going to use any other ctx, so feel free to override ctx, which is  regular pattern instead of creating "cc"